src/coreclr/vm/crst.h

```diff
@@ -352,17 +352,40 @@ class CrstBase
         CrstBase * m_pCrst;
 
     public:
-        inline CrstHolder(CrstBase * pCrst)
-            : m_pCrst(pCrst)
+        CrstHolder(CrstBase* pCrst)
+            : m_pCrst{ pCrst }
         {
             WRAPPER_NO_CONTRACT;
             AcquireLock(pCrst);
         }
 
-        inline ~CrstHolder()
+        ~CrstHolder()
         {
             WRAPPER_NO_CONTRACT;
-            ReleaseLock(m_pCrst);
+            if (m_pCrst != nullptr)
+                ReleaseLock(m_pCrst);
+        }
+
+        CrstHolder(CrstHolder const&) = delete;
+        CrstHolder &operator=(CrstHolder const&) = delete;
```

Comment:
This looks like a re-implementation of `CrstHolderWithState` below. I would leave plain `CrstHolder` alone (see command "CrstHolder is optimized for the common use that takes the lock in constructor..." above). Change CrstHolderWithState to do what you need if it does not do it already.

Reply:
I don't need this anymore actually. I'm going to make an explicit delete of the copy operators in question though. The rules for when they are implicitly marked delete is confusing.